src/us/kbase/workspace/database/ObjectIdentifier.java

```diff
@@ -87,6 +89,18 @@ public ObjectIdentifier(ObjectIdentifier id) {
 		this.id = id.id;
 		this.version = id.version;
 	}
+	
+	private ObjectIdentifier(
+			final WorkspaceIdentifier wsi,
+			final long id,
+			final String name,
+			final int version) {
+		this.wsi = wsi;
+		this.name = name;
+		// TODO NOW get rid of these ternaries
```

Comment:
Once these are done: https://github.com/kbase/workspace_deluxe/pull/508/files#diff-b848910f68b4a3592fc76ffa5e6e61960f66656e17e8d96147ff791177839329R26-R27

...the ternaries will be unnecessary, since the builder and the constructor types will match. I suppose these TODOs are redundant to the linked ones.